custom_components/tahoma/entity.py

```diff
@@ -136,3 +141,43 @@ def device_state_attributes(self) -> Dict[str, Any]:
                     attr[state.name] = state.value
 
         return attr
+
+
+@dataclass
+class OverkizSensorDescription(SensorEntityDescription):
+    """Class to describe a Overkiz sensor."""
+
+    value: Callable[[Any], Any] | None = lambda val: val
+
+
+@dataclass
+class OverkizBinarySensorDescription(BinarySensorEntityDescription):
+    """Class to describe a Overkiz binary sensor."""
+
+    is_on: Callable[[Any], Any] = lambda state: state
+
+
+class OverkizDescriptiveEntity(OverkizEntity):
+    """Representation of a Overkiz device entity based on a description."""
+
+    def __init__(
+        self,
+        device_url: str,
+        coordinator: TahomaDataUpdateCoordinator,
+        description: OverkizSensorDescription | OverkizBinarySensorDescription,
+    ):
+        """Initialize the device."""
+        super().__init__(device_url, coordinator)
+        self.entity_description = description
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        if self.executor.index:
+            return f"{self.entity_description.name} {self.executor.index}"
+        return self.entity_description.name
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return f"{super().unique_id}-{self.entity_description.key}"
```

Comment:
@tetienne my only concern is this one, do we need to add the platform here as well? (thus moving unique id to binary_sensory.py and sensor.py)?

I don't think that it wil happen that a sensor and binary sensor share the same state, but maybe I am overlooking anything?

Reply:
I think we are safe. The state name is unique among a device.